docs/standard/native-interop/abi-support.md

```diff
@@ -0,0 +1,36 @@
+---
+title: Native interoperability ABI support - .NET
+description: Defines the current support for interoperability with various ABIs.
+ms.date: 10/25/2024
+---
+# Native interoperability ABI support
+
+[Application Binary Interface](https://wikipedia.org/wiki/Application_binary_interface) (ABI) is the interface which run times and operating systems express low-level binary details. This can include calling conventions (that is, how parameters are passed and results returned), exception handling, as well as symbol mangling. The below list contains the names of languages, run times and even general technologies since these names tend to represent what users will be working with when looking for guidance on interoperability.
+
+## Support table
+
+|   | Support | APIs |
+|---|---------|------|
+| **C**                  | ✔️ | <xref:System.Runtime.InteropServices.LibraryImportAttribute> provides source generated support in .NET 7+. Use <xref:System.Runtime.InteropServices.DllImportAttribute> when targeting earlier .NET versions. Refer to [Interop best practices](./best-practices.md) for additional guidance. |
```

Comment:
People are likely to land on this page to answer "how to do interop between .NET and other languages". I think the "Support" column is confusing, in particular for C++. It will make people think that it is not possible to do interop between .NET and C++.

I liked he earlier formatting where each language had an explanatory paragraph better.

Reply:
> I think the "Support" column is confusing, in particular for C++.

Agreed. That is why I put "Guidance" on the API level. We technically don't interop with C++ is the issue. We don't have exception support any of the RAII mechanisms or even mangling resolution. Exporting via C is how it is done and that is not C++. I completely understand the nuance you are mentioning here though.

> I liked he earlier formatting where each language had an explanatory paragraph better.

Okay. I will convert back to that format instead.